ticdc: add a faq about generated column (#19035)
pingcap/docs-cn: ticdc/ticdc-faq.md

  }
]
```

## DML 中的生成列会发生什么行为？

> 生成列包括存储生成列和虚拟生成列。TICDC 会忽略虚拟生成列，而仅复制存储生成列到下游。当下游是 MySQL 数据库或其他与 MySQL 兼容的数据库，而不是 Kakfa 或其他存储服务时，存储生成列也会被忽略。

Reviewer: ```suggestion
生成列包括虚拟生成列和存储生成列。TiCDC 会忽略虚拟生成列，而仅同步存储生成列到下游。当下游是 MySQL 或其他与 MySQL 兼容的数据库，而不是 Kakfa 或其他存储服务时，存储生成列也会被忽略。
```

> > **Note:**
> >
> > 当复制存储生成列到 Kafka 或存储服务后，再写回 MySQL 时，可能会发生错误 `Error 3105 (HY000): The value specified for generated column 'xx' in table 'xxx' is not allowed`。您可以使用[Open Protocol]（https://docs.pingcap.com/zh/tidb/stable/ticdc-open-protocol#ticdc-open-protocol）来避免此错误，因为该协议携带 column flag, 可以区分是否为生成列。

Reviewer: ```suggestion
> **注意：**
>
> 当同步存储生成列到 Kafka 或存储服务后，再将其写回 MySQL 时，可能会遇到 `Error 3105 (HY000): The value specified for generated column 'xx' in table 'xxx' is not allowed` 错误。为避免该错误，你可以使用 [Open Protocol](/ticdc/ticdc-open-protocol.md#ticdc-open-protocol) 进行同步。该协议携带列的 flag 值，可以区分是否为生成列。
```

> > 当同步存储生成列到 Kafka 或存储服务后，再将其写回 MySQL 时，可能会遇到 `Error 3105 (HY000): The value specified for generated column 'xx' in table 'xxx' is not allowed` 错误。为避免该错误，你可以使用 [Open Protocol](/ticdc/ticdc-open-protocol.md#ticdc-open-protocol) 进行同步。该协议的输出包含[列的 flag 值](/ticdc/ticdc-open-protocol.md#列标志位)，可以区分是否为生成列。

Reviewer: ```suggestion
> 当同步存储生成列到 Kafka 或存储服务后，再将其写回 MySQL 时，可能会遇到 `Error 3105 (HY000): The value specified for generated column 'xx' in table 'xxx' is not allowed` 错误。为避免该错误，你可以使用 [Open Protocol](/ticdc/ticdc-open-protocol.md) 进行同步。该协议的输出包含[列的 flag 值](/ticdc/ticdc-open-protocol.md#列标志位)，可以区分是否为生成列。
```
